packages/components/src/components/SquiggleViewer/ViewerProvider.tsx

```diff
@@ -362,9 +361,54 @@ export function useViewerType() {
   return viewerType;
 }
 
+// List of callbacks that should do something outside of the viewer.
+// The common case is to scroll the editor to the necessary position in the playground.
+export type ExternalViewerActions = Partial<{
+  // TODO: this function is not imports-friendly yet.
+  // E.g. in stacktraces, the user might want to click on an error location and
+  // go to the source that's identified not just by an offset, but also its
+  // sourceId.
+  show: (location: SqLocation, focus: boolean) => void;
+  // When the viewer can focus on the location, will it be able to do that?
+  // This affects, for example, whether the links in stacktraces are clickable.
+  isFocusable: (location: SqLocation) => boolean;
+  // Should the viewer omit the source id because it's trivial?
+  // E.g. in playground stacktraces we want to show source id only for imports, but not for the currently edited file.
+  isDefaultSourceId: (sourceId: string) => boolean;
+}>;
+
+// Configure external actions for the common case of editor actions, e.g. for the playground or `<SquiggleEditor>` component.
+export function useExternalViewerActionsForEditor(
```

Comment:
I want to split `ViewerProvider` into multiple files, but didn't want to complicate the diff for this PR.

Reply:
Makes sense